packages/dds/tree/src/simple-tree/arrayNode.ts

```diff
@@ -620,8 +624,8 @@ function createArrayNodeProxy(
 				return Reflect.set(dispatch, key, newValue, proxy);
 			}
 
-			const field = getSequenceField(proxy);
-			const maybeIndex = asIndex(key, field.length);
+			// JS allows arrays to be set an any index, even an index that is arbitrarily past what is currently the end of the array.
```

Comment:
```suggestion
			// This type treats all non-negative integer indexes as array access. Use Infinity not length to ensure that indexing past the end doesn't create additional session local properties. 
```

Reply:
Also the "get" code should probably work similar, since we don't want to point out of bounds indexes to the additional properties: that just seems strange and should probably error as it won't have the correct type otherwise.

Reply:
Huh. What about "has"? Same thing?

I'm going to avoid changing any of the "read" behavior in this PR, but we should figure out what to do and fix it in a followup. Writing proxies is hard, isn't it... :/

Reply:
I've filed task 7753 to track this.